### ISSUE TITLE
Logging dependencies

### ISSUE THREAD

**Issue by rocketraman** [wildbit/postmark-java#27]
This library's logging dependencies need work. Here is the dependency report from Gradle:

```
+--- com.wildbit.java:postmark:1.7
|    +--- org.glassfish.jersey.core:jersey-client:2.33
|    |    +--- jakarta.ws.rs:jakarta.ws.rs-api:2.1.6
|    |    +--- org.glassfish.jersey.core:jersey-common:2.33
|    |    |    +--- jakarta.ws.rs:jakarta.ws.rs-api:2.1.6
|    |    |    +--- jakarta.annotation:jakarta.annotation-api:1.3.5
|    |    |    +--- org.glassfish.hk2.external:jakarta.inject:2.6.1
|    |    |    \--- org.glassfish.hk2:osgi-resource-locator:1.0.3
|    |    \--- org.glassfish.hk2.external:jakarta.inject:2.6.1
|    +--- org.glassfish.jersey.inject:jersey-hk2:2.33
|    |    +--- org.glassfish.jersey.core:jersey-common:2.33 (*)
|    |    +--- org.glassfish.hk2:hk2-locator:2.6.1
|    |    |    +--- org.glassfish.hk2.external:jakarta.inject:2.6.1
|    |    |    +--- org.glassfish.hk2.external:aopalliance-repackaged:2.6.1
|    |    |    +--- org.glassfish.hk2:hk2-api:2.6.1
|    |    |    |    +--- org.glassfish.hk2.external:jakarta.inject:2.6.1
|    |    |    |    +--- org.glassfish.hk2:hk2-utils:2.6.1
|    |    |    |    |    \--- org.glassfish.hk2.external:jakarta.inject:2.6.1
|    |    |    |    \--- org.glassfish.hk2.external:aopalliance-repackaged:2.6.1
|    |    |    \--- org.glassfish.hk2:hk2-utils:2.6.1 (*)
|    |    \--- org.javassist:javassist:3.25.0-GA
|    +--- javax.activation:activation:1.1.1
|    +--- com.fasterxml.jackson.core:jackson-core:2.10.0 -> 2.11.4
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.10.0 -> 2.11.4
|    +--- com.fasterxml.jackson.core:jackson-databind:2.10.0 -> 2.11.4 (*)
|    +--- eu.medsea.mimeutil:mime-util:2.1.3
|    |    +--- org.slf4j:slf4j-api:1.5.6 -> 1.7.30
|    |    +--- org.slf4j:slf4j-log4j12:1.5.6
|    |    |    +--- org.slf4j:slf4j-api:1.5.6 -> 1.7.30
|    |    |    \--- log4j:log4j:1.2.14
|    |    \--- log4j:log4j:1.2.14
|    +--- org.apache.logging.log4j:log4j-api:2.9.1 -> 2.14.1
|    \--- org.apache.logging.log4j:log4j-core:2.9.1 -> 2.14.1 (*)
```

I see a few issues here:

1) You should not expose log4j-core as a transitive dependency. Depending on `log4j-api` isn't too bad as at least it's just an API and not a logging implementation, but probably more standard for a library like this would be a dependency on slf4j-api or just use JUL.

2) Your `eu.medsea.mimeutil:mime-util` dependency transitively exposes an slf4j implementation `slf4j-log4j12` as well as a log4j implementation `log4j:log4j:1.2.14` (!). This is (very) wrong and until this is fixed upstream, you should exclude both of these transitive dependencies in your library so that projects including postmark-java don't get them transitively either.

3) As per https://github.com/wildbit/postmark-java/issues/26, I do note your library includes a `log4j2.properties` file. It should not -- configuring logging is a consuming application concern, not a library concern.

For Gradle, one way to work around 1 and 2 above is to exclude these dependencies in consuming projects, but this should not be necessary:

```
implementation("com.wildbit.java:postmark:1.7") {
  exclude("org.slf4j", "slf4j-log4j12")
  exclude("log4j", "log4j")
  exclude("org.apache.logging.log4j", "log4j-core")
}
```

**Comment by DarkNormal** [comment]
@rocketraman Thanks for the in-depth analysis of this, I was thinking my implementation was wrong initially when using this library. I raised #26 and had to apply some exclusions to the postmark dependency to get it working and maintain our current logging setup, as well as modifying the jar to remove the log4j2.properties file.

**Comment by ibalosh** [comment]
Thank you guys for the report, will be checking out these soon

**Comment by ibalosh** [comment]
Hi @rocketraman @DarkNormal ,

thank you for the detailed report, we have just released a version 1.7.2 which resolves these issues. All issues above should be resolved. log4j is not part of the library anymore since as you mentioned it's not really needed in this case and the mimeutil library is old one and we went for newer library, for customers wanting to use auto mime file type detection.

**Comment by rocketraman** [comment]
Thank you @ibalosh , I can confirm the situation with logging is much better now.

However, why do you depend on tika-bundle instead of tika-core directly?

```
+--- com.wildbit.java:postmark:1.7.2
...
|    \--- org.apache.tika:tika-bundle:1.25
|         \--- org.apache.tika:tika-core:1.25
```

The `tika-bundle` dependency bundles all the tika dependencies together (everything needed for everything Tika does, including parsing a bazillion different file types) and adds OSGi meta-data on top, so that tika can be used easily in an OSGi environment. As a result, it is quite large (48 Megabytes).

For MIME type detection, you should only need tika-core (712 Kb) and nothing else.

I can open a separate issue for this if you like.

**Comment by ibalosh** [comment]
Hi @rocketraman 
no need, already addressed in 1.7.3
